src/Domain/Model/WinningLotto.js

```diff
@@ -0,0 +1,27 @@
+import Lotto from './Lotto.js';
+
+class WinningLotto extends Lotto {
+  #bonusNumber;
+
+  constructor(numbers, bonusNumber) {
+    super(numbers);
+    this.#bonusNumber = bonusNumber;
+  }
+
+  getBonusNumber() {
+    return this.#bonusNumber;
+  }
+
+  countMatchingNumbers(lotto) {
+    const matchingNumbers = this.getNumbers().filter((number) =>
+      lotto.getNumbers().includes(number)
+    ).length;
+    return matchingNumbers;
+  }
```

Comment:
```suggestion
    countMatchingNumbers(lotto) {
      const winningNumbersSet = new Set(this.getNumbers());
      return lotto.getNumbers().filter(number => winningNumbersSet.has(number)).length;
    }
```
성능과 가독성을 위해 이렇게도 개선할 수 있을 것 같은데 어떻게 생각하세요??

Reply:
오 코드가 더 간결하고 보기 좋아졌네요!! 

한가지 궁금한 점은 꼭 set()을 해줘야할까요? 저는 Set()을 사용하는 목적은 중복 제거라고 생각했는데요. 
이미 winningNumber의 경우, 유효성 검사를 마친 중복없는 숫자들을 갖고 있기에 꼭 필요한 과정은 아니라고 생각했습니다. 그런데 countMatchingNumbers() 내부에서 Set()을 사용해서 winningNumbersSet을 구하는 한 줄의 이유가 궁금합니다!

```js
countMatchingNumbers(lotto) {
      return lotto.getNumbers().filter(number => this.getNumbers().includes(number)).length;
    }
```

흠 이렇게 쓰고 보니 가독성면에서도 떨어지네요 !

게다가 Array.includes() 와 Set.has()를 비교했을때 시간복잡도 차이가 O(N)과 O(1)로 크네요 ..! 로또 번호가 6개라서 큰 성능 차이는 없겠지만 데이터가 많아지면 Set을 쓰는게 훨씬 낫겠군요

반영해서 수정했습니다! (commit : [b1eb175](https://github.com/woowacourse/javascript-lotto/pull/370/commits/b1eb175e710bf3bc5047db88f564ffe5a1690284))

Reply:
아 사실 winningNumber가 유효성검사를 마친 중복없는 숫자들을 갖고 있다는 사실을 인지하지 못했어요. 이렇게 validation이 도메인이 아닌 view에 위치하게 되면 관리가 어려워지게 되는 것 같아요. 도메인에 종속되는 validation이라면 도메인에 위치시키는 것은 어떨까요? 만약에 또다른 view에서 동일한 Lotto객체를 사용하게 될 때 또다시 validation로직을 중복으로 적어줘야하잖아요? Lotto에 validation책임도 추가되면 어떨지 제안드려봅니다!